openverse_catalog/dags/providers/provider_api_scripts/phylopic.py

```diff
@@ -13,272 +13,282 @@
-    """
-    This script pulls the data for a given date from the PhyloPic
-    API, and writes it into a .TSV file to be eventually read
-    into our DB.
-
-    Required Arguments:
-
-    date_start:  Date String in the form YYYY-MM-DD. This date defines the beginning
-                 of the range that the script will pull data from.
-    date_end:    Date String in the form YYYY-MM-DD. Similar to `date_start`, this
-                 defines the end of the range of data pulled. Defaults to
-                 DEFAULT_PROCESS_DAYS (7) if undefined.
-    """
-
-    offset = 0
-
-    logger.info("Begin: PhyloPic API requests")
-
-    if date_start == "all":
-        logger.info("Processing all images")
-        param = {"offset": offset}
-
-        image_count = _get_total_images()
-        logger.info(f"Total images: {image_count}")
-
-        while offset <= image_count:
-            _add_data_to_buffer(**param)
-            offset += LIMIT
-            param = {"offset": offset}
-
-    else:
-        if date_end is None:
-            date_end = _compute_date_range(date_start)
-        param = {"date_start": date_start, "date_end": date_end}
-
-        logger.info(f"Processing from {date_start} to {date_end}")
-        _add_data_to_buffer(**param)
-
-    image_store.commit()
-
-    logger.info("Terminated!")
-
-
-def _add_data_to_buffer(**kwargs):
-    endpoint = _create_endpoint_for_IDs(**kwargs)
-    IDs = _get_image_IDs(endpoint)
-    if IDs is None:
-        return
-
-    for id_ in IDs:
-        item_data = _get_object_json(id_)
-        if item_data:
-            _process_item(item_data)
-
-
-def _process_item(item_data: Dict) -> None:
-    details = _get_meta_data(item_data)
-    if details is not None:
-        image_store.add_item(**details)
-
-
-def _get_total_images() -> int:
-    # Get the total number of PhyloPic images
-    total = 0
-    endpoint = "http://phylopic.org/api/a/image/count"
-    result = delayed_requester.get_response_json(endpoint, retries=2)
-
-    if result and result.get("success") is True:
-        total = result.get("result", 0)
-
-    return total
-
-
-def _create_endpoint_for_IDs(**kwargs):
-    limit = LIMIT
-
-    if ((date_start := kwargs.get("date_start")) is not None) and (
-        (date_end := kwargs.get("date_end")) is not None
+class PhylopicDataIngester(ProviderDataIngester):
+    delay = 5
+    host = "http://phylopic.org"
+    endpoint = f"{host}/api/a/image"
+    providers = {"image": prov.PHYLOPIC_DEFAULT_PROVIDER}
+    batch_limit = 25
+    # Default number of days to process if date_end is not defined
+    default_process_days = 7
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        # This is made an instance attribute rather than passed around via query params
+        # because it actually comprises the URL we hit (not the params) depending on
+        # whether we're running a dated DAG or not.
```

Comment:
Nit: I think the word comprise is not used correctly here. This confused me for a bit (had to look up the word to make sure I was understanding correctly). Is this more accurate?

```suggestion
        # This is made an instance attribute rather than passed around via query params
        # because the URL we hit includes it in the path (not the params) depending on
        # whether we're running a dated DAG or not.
```